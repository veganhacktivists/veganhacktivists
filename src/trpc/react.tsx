'use client';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { loggerLink, unstable_httpBatchStreamLink } from '@trpc/client';
import { createTRPCReact } from '@trpc/react-query';
import { type inferRouterInputs, type inferRouterOutputs } from '@trpc/server';
import { useState } from 'react';
import superjson from 'superjson';
import { Prisma } from '@prisma/client';

import type { AppRouter } from 'server/api/root';

const createQueryClient = () => new QueryClient();

let clientQueryClientSingleton: QueryClient | undefined = undefined;
const getQueryClient = () => {
  if (typeof window === 'undefined') {
    // Server: always make a new query client
    return createQueryClient();
  }
  // Browser: use singleton pattern to keep the same query client
  return (clientQueryClientSingleton ??= createQueryClient());
};

export const api = createTRPCReact<AppRouter>();

/**
 * Inference helper for inputs.
 *
 * @example type HelloInput = RouterInputs['example']['hello']
 */
export type RouterInputs = inferRouterInputs<AppRouter>;

/**
 * Inference helper for outputs.
 *
 * @example type HelloOutput = RouterOutputs['example']['hello']
 */
export type RouterOutputs = inferRouterOutputs<AppRouter>;

superjson.registerCustom<Prisma.Decimal, string>(
  {
    isApplicable: (v): v is Prisma.Decimal => Prisma.Decimal.isDecimal(v),
    serialize: (v) => v.toJSON(),
    deserialize: (v) => new Prisma.Decimal(v),
  },
  'decimal.js',
);

export const TRPCReactProvider = (props: { children: React.ReactNode }) => {
  const queryClient = getQueryClient();

  const [trpcClient] = useState(() =>
    api.createClient({
      links: [
        loggerLink({
          enabled: (op) =>
            process.env.NODE_ENV === 'development' ||
            (op.direction === 'down' && op.result instanceof Error),
        }),
        unstable_httpBatchStreamLink({
          transformer: superjson,
          url: getBaseUrl() + '/api/trpc',
          headers: () => {
            return { 'x-trpc-source': 'nextjs-react' };
          },
        }),
      ],
    }),
  );

  return (
    <QueryClientProvider client={queryClient}>
      <api.Provider client={trpcClient} queryClient={queryClient}>
        {props.children}
      </api.Provider>
    </QueryClientProvider>
  );
};

function getBaseUrl() {
  if (typeof window !== 'undefined') return window.location.origin;
  if (process.env.URL) return `https://${process.env.URL}`;
  return `http://localhost:${process.env.PORT ?? 3000}`;
}
