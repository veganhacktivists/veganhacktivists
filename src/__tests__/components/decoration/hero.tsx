/* eslint-disable react/display-name */
import React from 'react';
import { render } from '@testing-library/react';
import Hero from '../../../components/decoration/hero';

jest.mock(
  'next/image',
  () =>
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    function ({ src, alt }: any) {
      // eslint-disable-next-line @next/next/no-img-element
      return <img src={src} alt={alt} />;
    }
);

it('should render correctly when providing a URL string', () => {
  const { asFragment } = render(
    <Hero
      imageBackground="https://via.placeholder.com/500x500"
      alignment="right"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should render correctly when providing static image data', () => {
  const { asFragment } = render(
    <Hero
      imageBackground={{
        src: 'https://via.placeholder.com/500x500',
        width: 500,
        height: 500,
      }}
      alignment="left"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
