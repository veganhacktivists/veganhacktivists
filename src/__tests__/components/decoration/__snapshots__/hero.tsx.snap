// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when providing a URL string 1`] = `
<DocumentFragment>
  <div
    class="relative md:-top-20 md:-mb-20 p-10 md:p-32 bg-scroll bg-cover bg-no-repeat flex justify-end"
  >
    <img
      alt=""
      src="https://via.placeholder.com/500x500"
    />
    <div
      class="flex flex-col justify-center w-1/2"
    />
    <div
      class="absolute inset-0 overflow-hidden"
    >
      <svg
        class="text-white overflow-visible absolute top-0 right-0 transform translate-x-1/4 -translate-y-2/4 lg:-translate-y-3/4
    "
        height="60vw"
        width="60vw"
      >
        <circle
          class="stroke-current"
          cx="30vw"
          cy="30vw"
          fill-opacity="0"
          opacity="0.1"
          r="30vw"
          stroke-width="20px"
        />
      </svg>
      <svg
        class="text-white overflow-visible absolute bottom-0 left-0 transform -translate-x-1/4 translate-y-2/4 lg:translate-y-3/4
    "
        height="68vw"
        width="68vw"
      >
        <circle
          class="stroke-current"
          cx="34vw"
          cy="34vw"
          fill-opacity="0"
          opacity="0.5"
          r="34vw"
          stroke-width="20px"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render correctly when providing static image data 1`] = `
<DocumentFragment>
  <div
    class="relative md:-top-20 md:-mb-20 p-10 md:p-32 bg-scroll bg-cover bg-no-repeat flex justify-start"
  >
    <img
      alt=""
      src="[object Object]"
    />
    <div
      class="flex flex-col justify-center w-1/2"
    />
    <div
      class="absolute inset-0 overflow-hidden"
    >
      <svg
        class="text-white overflow-visible absolute top-0 right-0 transform translate-x-1/4 -translate-y-2/4 lg:-translate-y-3/4
    "
        height="60vw"
        width="60vw"
      >
        <circle
          class="stroke-current"
          cx="30vw"
          cy="30vw"
          fill-opacity="0"
          opacity="0.1"
          r="30vw"
          stroke-width="20px"
        />
      </svg>
      <svg
        class="text-white overflow-visible absolute bottom-0 left-0 transform -translate-x-1/4 translate-y-2/4 lg:translate-y-3/4
    "
        height="68vw"
        width="68vw"
      >
        <circle
          class="stroke-current"
          cx="34vw"
          cy="34vw"
          fill-opacity="0"
          opacity="0.5"
          r="34vw"
          stroke-width="20px"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
